scan/base.go

```diff
@@ -40,6 +42,303 @@ type base struct {
+		err = fmt.Errorf("not found : WpToken")
+		return
+	}
+
+	cmd := []Command{{Command: "wp cli", Name: "wp"}, {Command: "curl --help", Name: "curl"}}
+	for _, i := range cmd {
+		if r := exec(l.ServerInfo, i.Command, noSudo); !r.isSuccess() {
+			err = fmt.Errorf("%s command not installed", i.Name)
+			return
+		}
+	}
+
+	var unsecures []models.VulnInfo
+	if unsecures, err = detectWp(l); err != nil {
+		l.log.Errorf("Failed to scan wordpress: %s", err)
+		return err
+	}
+	l.WpVulnInfos = map[string]models.VulnInfo{}
+	for _, i := range unsecures {
+		l.WpVulnInfos[i.CveID] = i
+	}
+
+	return
+}
+
+//WpCveInfos is for wpvulndb's json
+type WpCveInfos struct {
+	ReleaseDate     string      `json:"release_date"`
+	ChangelogURL    string      `json:"changelog_url"`
+	Status          string      `json:"status"`
+	LatestVersion   string      `json:"latest_version"`
+	LastUpdated     string      `json:"last_updated"`
+	Popular         bool        `json:"popular"`
+	Vulnerabilities []WpCveInfo `json:"vulnerabilities"`
+	Error           string      `json:"error"`
+}
+
+//WpCveInfo is for wpvulndb's json
+type WpCveInfo struct {
+	ID            int        `json:"id"`
+	Title         string     `json:"title"`
+	CreatedAt     string     `json:"created_at"`
+	UpdatedAt     string     `json:"updated_at"`
+	PublishedDate string     `json:"Published_date"`
+	VulnType      string     `json:"Vuln_type"`
+	References    References `json:"references"`
+	FixedIn       string     `json:"fixed_in"`
+}
+
+//References is for wpvulndb's json
+type References struct {
+	URL     []string `json:"url"`
+	Cve     []string `json:"cve"`
+	Secunia []string `json:"secunia"`
+}
+
+func detectWp(c *base) (vinfos []models.VulnInfo, err error) {
+
+	var coreVuln []models.VulnInfo
+	if coreVuln, err = detectWpCore(c); err != nil {
+		return
+	}
+	for _, i := range coreVuln {
+		vinfos = append(vinfos, i)
+	}
+
+	var themeVuln []models.VulnInfo
+	if themeVuln, err = detectWpTheme(c); err != nil {
+		return
+	}
+	for _, i := range themeVuln {
+		vinfos = append(vinfos, i)
+	}
+
+	var pluginVuln []models.VulnInfo
+	if pluginVuln, err = detectWpPlugin(c); err != nil {
+		return
+	}
+	for _, i := range pluginVuln {
+		vinfos = append(vinfos, i)
+	}
+
+	return
+}
+
+func detectWpCore(c *base) (vinfos []models.VulnInfo, err error) {
+	cmd := fmt.Sprintf("wp core version --path=%s", c.ServerInfo.WpPath)
+
+	var coreVersion string
+	if r := exec(c.ServerInfo, cmd, noSudo); r.isSuccess() {
+		tmp := strings.Split(r.Stdout, ".")
+		coreVersion = strings.Join(tmp, "")
+		coreVersion = strings.TrimRight(coreVersion, "\r\n")
+		if len(coreVersion) == 0 {
+			return
+		}
+	}
+	cmd = fmt.Sprintf("curl -k -H 'Authorization: Token token=%s' https://wpvulndb.com/api/v3/wordpresses/%s", c.ServerInfo.WpToken, coreVersion)
+	if r := exec(c.ServerInfo, cmd, noSudo); r.isSuccess() {
+		coreConvertVinfo(r.Stdout)
+	}
+	return
+}
+
+func coreConvertVinfo(stdout string) (vinfos []models.VulnInfo, err error) {
```

Comment:
what is stdout?
Variable names are more readable if they write as concretely as possible